### DISCUSSION TITLE
Disable TOC via plugin settings

### DISCUSSION THREAD

**Discussion by TheZoker** [facebook/docusaurus#5220] [Ideas]
Hi there,

I know that I can disable the TOC for every page by setting `hide_table_of_contents: true` but it would be nice, if this was also a plugin setting like:
```js
module.exports = {
  plugins: [
    [
      '@docusaurus/plugin-content-docs',
      {
         ...
         disableToc: true,
         ...
      }
    ]
  ]
}
```

That would allow me to disable TOC per docs instance and not having to disable it for every item specifically. 

Thanks!

**Comment by slorber** [comment]
What I'd like to have is:
- 3 frontmatter: `hide_table_of_contents` + `hide_table_of_contents_desktop` + `hide_table_of_contents_mobile` for better granularity
- a plugin config that allows you to modify the frontmatter on a per-doc basis, so that we don't need to add plugin options for providing defaults

You would use it like that for your use-case:

```js
docs: {
  createDocFrontMatter: ({doc,frontMatter)}) => ({
    ...frontMatter,
    "hide_table_of_contents": true,
  }),
}
```

Does it make sense?

This generic system would be more powerful to support other use-cases as well, and would enable to turn on/off features like TOC on a per-doc basis. For example you could express that you want docs in a certain folder to have TOC, but not docs in another folder (instead of a static per-instance config)

**Reply by TheZoker** [reply]
Sounds good, although it seems a bit more complicated then my proposal. But as long as I'm able to disable the TOC on a docs instance level (have it enabled for one and disable it for another) it works for me 👍

**Comment by pelikhan** [comment]
It would be nice to be able to provide default front_matter values, specially to force `hide_table_of_contents` to true

**Reply by slorber** [reply]
Agree, but this will also be possible with the solution mentioned in https://github.com/facebook/docusaurus/discussions/5220#discussioncomment-1061122
